src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,71 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String PARSED_STRING_SEPARATOR = ";";
+    private static final String FORMED_STRING_SEPARATOR = ",";
+    private static final String BUY_CATEGORY_NAME = "buy";
+    private static final String SUPPLY_CATEGORY_NAME = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = readDataFromFile(fromFileName);
+        String report = formReport(data);
+        writeFile(toFileName, report);
+    }
+
+    public String[] readDataFromFile(String fromFileName) {
+        StringBuilder parsedCsv = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String value;
+            while ((value = reader.readLine()) != null) {
+                parsedCsv.append(value).append(PARSED_STRING_SEPARATOR);
+            }
+
+            return parsedCsv.toString().split(PARSED_STRING_SEPARATOR);
+        } catch (Exception e) {
+            throw new RuntimeException("Can't read data from the file " + fromFileName, e);
+        }
+    }
+
+    private String formReport(String[] parsedString) {
+        int buyAmount = 0;
+        int supplyAmount = 0;
+        StringBuilder result = new StringBuilder();
+
+        for (String data: parsedString) {
+            int amountNumber = Integer.parseInt(data.split(",")[1]);
+            if (data.contains(BUY_CATEGORY_NAME)) {
+                buyAmount += amountNumber;
+            } else if (data.contains(SUPPLY_CATEGORY_NAME)) {
+                supplyAmount += amountNumber;
+            }
```

Comment:
```suggestion
            String records = data.split(FORMED_STRING_SEPARATOR);
            int amountNumber = Integer.parseInt(records[OPERATION_AMOUNT_INDEX]);
            if (records[OPERATION_INDEX].equels(BUY_CATEGORY_NAME)) {
                buyAmount += amountNumber;
            } else  {
                supplyAmount += amountNumber;
            }
```
I meant to do something like this. It increases productivity in your solution (why?)

Reply:
Increases readability - no magic numbers